public/app/plugins/panel/singlestat/module.ts

```diff
@@ -113,6 +114,32 @@ class SingleStatCtrl extends MetricsPanelCtrl {
     this.unitFormats = kbn.getUnitFormats();
   }
 
+  migrateToGaugePanel(migrate: boolean) {
+    if (migrate) {
+      console.log('Change the panel type... the wrapper should pick this up after render');
+      this.panel.type = 'gauge';
+
+      // maybe change some parameter?
+      getLocationSrv().update({ query: { chage: 'xxxxx' }, partial: true });
+
+      // or refresh?
+      this.refresh();
+
+      // debugger;
+      // const panel = this.panel as PanelModel;
+      // importPanelPlugin('gauge').then( gauge => {
```

Comment:
this does not work... we need the DashboardPanel to call its update function

I'm not sure how to best get an event there